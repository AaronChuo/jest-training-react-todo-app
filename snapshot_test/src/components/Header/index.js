import React, { PropTypes, Component } from 'react';
import TodoTextInput from 'components/TodoTextInput';
import style from 'todomvc-app-css/index.css';

export default class Header extends Component {
  static propTypes = {
    addTodo: PropTypes.func.isRequired,
  };

  handleSave = (text = '') => {
    if (text.length !== 0) {
      this.props.addTodo({ text });
    }
  };

  render () {
    return (
      <header className={style['header']}>
        <h1>todos</h1>
        <TodoTextInput
          newTodo
          onSave={this.handleSave}
          placeholder="What needs to be done?"
        />
      </header>
    );
  }
}
