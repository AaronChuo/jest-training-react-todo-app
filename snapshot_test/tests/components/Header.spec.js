import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Header from 'components/Header';
import TodoTextInput from 'components/TodoTextInput';

describe('<Header />', () => {
  let handleAddTodo;
  let wrapper;

  beforeEach(() => {
    handleAddTodo = jest.fn();
    wrapper = shallow(
      <Header
        addTodo={handleAddTodo}
      />
    );
  });

  it('should render initial component', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('simulates save todo', () => {
    const mockTodoText = 'Go shopping';
    wrapper.find(TodoTextInput).simulate('save', mockTodoText);
    expect(handleAddTodo).toHaveBeenCalledWith({ text: mockTodoText });
  });

  it('simulates save todo without text', () => {
    wrapper.find(TodoTextInput).simulate('save');
    expect(handleAddTodo).not.toHaveBeenCalled();
  });
});
