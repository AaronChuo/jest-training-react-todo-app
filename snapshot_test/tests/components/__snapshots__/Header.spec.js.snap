// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render initial component 1`] = `
<header
  className="header"
>
  <h1>
    todos
  </h1>
  <TodoTextInput
    newTodo={true}
    onSave={[Function]}
    placeholder="What needs to be done?"
  />
</header>
`;
